docs/adr/0007-seperate-binding-generators.md

```diff
@@ -0,0 +1,142 @@
+## Decision Drivers
+
+* Maintainability, we would like to improve the maintainability of our binding generators - this means clear ownership of the bindings.
+* Testability, it should be easy for developers to test the bindings they care about. Without having to navigate and install unfamiliar libraries and build systems.
+* Version compatibility, we need to guarantee that the same version of `uniffi` is used to generate both the scaffolding and the bindings. This is easy to ensure when all the bindings (and the scaffolding) live under the same crate but becomes a concern if they are separated.
+* Developer experience, it shouldn't be harder to write and maintain a new binding generator than it currently is.
+* Releases, cutting releases for changes in one binding generator shouldn't harm another.
+
+## Considered Options
+
+* **[Option 1] Do nothing**
+
+    This means keeping everything as-is, and deciding that all binding generators (at least for now) should live under the `uniffi_bindgen` crate.
+
+* **[Option 2] Expose a trait in `uniffi_bindgen` that can be implemented by other crates, to separate the implementation of the bindings**
+
+    This means that `uniffi_bindgen` would still handle generic tasks related to binding generation. This includes:
+        1. Parsing the UDL, and passing the `ComponentInterface` to the specific binding generator
+        1. Calling a generic `BindingGenerator::write_bindings` function that is implemented by the specific binding generator
+        1. `uniffi_bindgen` would also expose its `CodeOracle`, `CodeType` and `CodeDeclaration` types to help developers create a standard way to interact with code generation. It shouldn't however, restrict developers to use those concepts if they don't fit a specific binding.
+    But developers would need a separate crate that implements the traits `uniffi_bindgen` exposes.
+
+* **[Option 3] Remove `uniffi_bindgen`, replace it with `uniffi_scaffolding` and have each binding generator implementation fully manage the generation of bindings**
+    This means external crates would have to depend on `uniffi` directly and implement command-line interfaces. This gives full autonomy to the binding generator developer. Developers won't have access to any traits from `uniffi_bindgen`
+
+## Pros and Cons of the Options
+
+### **[Option 1] Do nothing**
+
+* Good, because it keeps version compatibility easy to guarantee.
+* Good, it makes it easier to make changes to multiple bindings at a time (in the case of a breaking change in `uniffi`, etc)
+* Bad, because maintainability can grow to be difficult - especially if more bindings are added which the core `uniffi` team is not familiar with
+* Bad, Because testability can also grow to be difficult - as more bindings are added (some type of `gecko-js` is expected in the future) the requirement to test all the bindings together in one repository is difficult to maintain.
```

Comment:
```suggestion
* Bad, because testability can also grow to be difficult - as more bindings are added (some type of `gecko-js` is expected in the future) the requirement to test all the bindings together in one repository is difficult to maintain.
```